src/lua/lua_task.c

```diff
@@ -2644,22 +2691,18 @@ lua_task_inject_url(lua_State *L)
 
 	if (lua_isuserdata(L, 3)) {
 		/* We also have a mime part there */
-		mpart = *((struct rspamd_mime_part **) rspamd_lua_check_udata_maybe(L,
-																			3, rspamd_mimepart_classname));
+		mpart = *((struct rspamd_mime_part **)
+							   rspamd_lua_check_udata_maybe(L,3,rspamd_mimepart_classname));
 	}
-
 	if (task && task->message && url && url->url) {
 		if (rspamd_url_set_add_or_increase(MESSAGE_FIELD(task, urls), url->url, false)) {
-			if (mpart && mpart->urls) {
-				/* Also add url to the mime part */
-				g_ptr_array_add(mpart->urls, url->url);
-			}
+			GPtrArray* part_urls = g_ptr_array_new();
+			inject_url_query(task, url->url, part_urls);
```

Comment:
I'm sorry but I don't understand that. Why don't we use `mpart->urls` here? We will clearly 1) miss urls from part; 2) leak memory.

Reply:
Because if maprt is NULL, it will cause a segmentation fault, so in order not to catch this error, we will need to create a fake mpart, but as mentioned earlier, we should not fake the mpart. Also, if I understand correctly, the mime part has already parsed all its URLs to `taks->urls`.

Reply:
No, if mpart is null, then you don't need to insert a new url there, that's it.